app/src/main/java/org/wikipedia/csrf/CsrfTokenClient.kt

```diff
@@ -29,13 +29,13 @@ object CsrfTokenClient {
                 val service = svc ?: ServiceFactory.get(site)
                 var lastError: Throwable? = null
                 for (retry in 0 until MAX_RETRIES) {
-                    if (retry > 0) {
+                    if (retry > 0 && AccountUtil.isLoggedIn && !AccountUtil.isTemporaryAccount) {
```

Comment:
I'm also adding an explicit guard for re-logging in only if it's definitely not a temp account.